src/materials/ConcreteASREigenstrain.C

```diff
@@ -0,0 +1,323 @@
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ConcreteASREigenstrain.h"
+
+registerMooseObject("BlackBearApp", ConcreteASREigenstrain);
+
+template <>
+InputParameters
+validParams<ConcreteASREigenstrain>()
+{
+  InputParameters params = validParams<ConcreteExpansionEigenstrainBase>();
+
+  params.addRequiredCoupledVar("temperature", "Coupled temperature");
+  params.addRequiredCoupledVar("relative_humidity", "Coupled relative humidity");
+
+  params.addRangeCheckedParam<Real>(
+      "rh_exponent",
+      4.0,
+      "rh_exponent >= 0.0",
+      "Power to which relative humidity is raised in computation of ASR volumetric strain");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "max_volumetric_expansion",
+      "max_volumetric_expansion > 0.0",
+      "Final ansymptotic ASR volumetric expansion strain when reaction is complete");
+  params.addRequiredRangeCheckedParam<Real>(
+      "characteristic_time",
+      "characteristic_time > 0.0",
+      "Chracteristic ASR time (in days) at reference temprature. (tau_C(T_0))");
+  params.addRequiredRangeCheckedParam<Real>(
+      "latency_time",
+      "latency_time > 0.0",
+      "Latency ASR time (in days) at reference temprature (tau_L(T_0))");
+  params.addRangeCheckedParam<Real>("characteristic_activation_energy",
+                                    5400.0,
+                                    "characteristic_activation_energy > 0.0",
+                                    "Activation energy associated with characteristic_time (U_C)");
+  params.addRangeCheckedParam<Real>("latency_activation_energy",
+                                    9400.0,
+                                    "latency_activation_energy > 0.0",
+                                    "Activation energy associated with latency_time (U_L)");
+  params.addRequiredParam<Real>("reference_temperature",
+                                "Reference temperature for ASR reaction constants.");
+
+  // Note that Fahrenheit is not supported because that would require different parameters for the
+  // times and activation energies
+  MooseEnum temperature_units("Celsius Kelvin");
+  params.addRequiredParam<MooseEnum>(
+      "temperature_unit",
+      temperature_units,
+      "Unit used to define 'temperature' and 'reference_temperature'");
+
+  params.addRangeCheckedParam<Real>(
+      "stress_latency_factor",
+      1.3333,
```

Comment:
Would a fraction be better than a truncated decimal? Or is this value from literature?

Reply:
It would be better. I'm pretty sure we can't put a mathematical operation here, though.

Reply:
Turns out you actually can have mathematical operations there.  Cool!